### ISSUE TITLE
🐞 dagger init should not create a LICENSE file in the typescript SDK

### ISSUE THREAD

**Issue by drmikecrowe** [dagger/dagger#8527]
### What is the issue?

Yes, I know https://github.com/dagger/dagger/issues/7703 was debated.  Not sure if the typescript SDK is not fixed yet or what.

Dagger is a build system for end-user projects.  You should never arbitrarily add a LICENSE file if one does not exist.

@helderco  -- [your comments](https://github.com/dagger/dagger/issues/7703#issuecomment-2181310539) specify the exact behavior end users expect:

* `dagger init --sdk ....` -- no license file created
* `dagger init --sdk ... --license` -- create a license file

I get the daggerverse angle, but you can't 

### Dagger version

dagger v0.13.3 (registry.dagger.io/engine:v0.13.3) linux/amd64

### Steps to reproduce

```
internal-dagger  dagger @ dd63e5a ⩮ ❯ dagger init --sdk=typescript --source=./.dagger
✔ connect 0.3s
✔ moduleSource(refString: "."): ModuleSource! 0.0s
✔ ModuleSource.kind: ModuleSourceKind! 0.0s
✔ ModuleSource.resolveContextPathFromCaller: String! 0.0s
✔ ModuleSource.withName(name: "internal-dagger"): ModuleSource! 0.0s
✔ ModuleSource.withSDK(sdk: "typescript"): ModuleSource! 0.0s
✔ ModuleSource.withInit: ModuleSource! 0.0s
✔ ModuleSource.withSourceSubpath(path: ".dagger"): ModuleSource! 0.0s
✔ ModuleSource.resolveFromCaller: ModuleSource! 1.1s
✔ ModuleSource.asModule(engineVersion: "latest"): Module! 47.7s
✔ Module.generatedContextDiff: Directory! 0.0s
✔ Directory.export(path: "/home/mcrowe/Programming/Pinnacle/internal-dagger"): String! 5.3s

07:19:41 WRN no LICENSE file found; generating one for you, feel free to change or remove license=Apache-2.0
Initialized module internal-dagger in .

internal-dagger  dagger @ dd63e5a ⩮⋄ 55s ❯ git status
On branch dagger
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        .dagger/
        LICENSE
        dagger.json
```

**Comment by helderco** [comment]
This happens in the CLI, for every SDK. There's nothing specific to TypeScript.

But what you ask is actually in alignment with:
- https://github.com/dagger/dagger/issues/8452
- https://github.com/dagger/dagger/issues/8511

I know this has been discussed internally in relation to the Daggerverse, but I don’t know if anything’s been decided yet. \cc @jpadams, @shykes

**Comment by IshaSri-17Speed** [comment]
Yes , I agree with @helderco , got the same warning while using go SDK ( common for every SDK) . 

**_12:46:58 WRN no LICENSE file found; generating one for you, feel free to change or remove license=Apache-2.0_** 


As Dagger generated a default LICENSE file  with the Apache-2.0 license , it's a common behavior when creating new projects or modules. 
What I'm debating is , can I safely remove this LICENSE file without any issues ?

**Comment by drmikecrowe** [comment]
> What I'm debating is , can I safely remove this LICENSE file without any issues ?

Yes -- the goal of dagger is to not litter _your_ source.  If the LICENSE was not present before dagger, you are safe to remove it

**Comment by IshaSri-17Speed** [comment]
**Thankyou @drmikecrowe for the confirmation** 

Can we   **Introduce a command-line flag (e.g., --no-license) that users can specify when running dagger init to skip generating the LICENSE file**.
Alternatively, moving the creation of the **`LICENSE`** file to the **`dagger develop`** command would ensure that it only appears when actual code is being generated as discussed in #6666.

Change would helps in  **streamlining  project initialization and reduce unnecessary clutter in repositories.**

**Comment by drmikecrowe** [comment]
These are my personal opinions as an end user:

* Bottom line: NEVER add a license to an end user's production code unless the user adds the `--license`.  Changing a proprietary project to open-source would be one of the worst things you could do (this is what happened to me).
* The `dagger init/develop --license` is the correct approach
* If you are trying to address Daggerverse modules, Daggerverse is broken beyond the LICENSE aspect and not very useful in the node/TypeScript world (sorry to say).  For example, [this kind of content](https://daggerverse.dev/search?q=npm) isn't very useful.  I can expand on this if you are interested

**Comment by helderco** [comment]
> Can we **Introduce a command-line flag (e.g., --no-license) that users can specify when running dagger init to skip generating the LICENSE file**. 

You can actually skip creating the license with an empty value to `--license`:
```
dagger init --sdk=go --license=
```

*(`--license=` is that same as `--license ""`)*

But I realize that this is neither intuitive nor documented in `--help`. The reason for this is in:
- https://github.com/dagger/dagger/pull/7719#discussion_r1649268408

Basically, we decided not to change the default at the time. That PR was fixing a regression where the LICENSE was being created **every** time, not just on init. But I suggested taking the opportunity to allow opting out so that users didn’t feel stuck with it. However, I failed to follow up with either docs or a better solution.

I think this issue does that. We should either:
1. Create a `--no-license` flag to prevent creating the LICENSE
2. Stop creating it by default, requiring an explicit `--license`

The second one is my preference:
- `dagger init --sdk=go` → don’t create license
- `dagger init --sdk=go --license` → create license, defaulting to Apache-2.0
- `dagger init --sdk=go --license=MIT` → create MIT license
- `dagger init --sdk=go --license=` → 🤷 empty | proprietary | no-op | error | use default

---

> Alternatively, moving the creation of the **`LICENSE`** file to the **`dagger develop`** command would ensure that it only appears when actual code is being generated as discussed in #6666.

Both `dagger init` and `dagger develp` can generate code, what triggers it is the `--sdk` flag:

```
dagger init
dagger develop --sdk=go
```

It’s just hat `dagger init` can do those steps in one go.

**Comment by drmikecrowe** [comment]
I'm struggling with why you are trying to create a LICENSE file.  

Here's my new-user view of my dagger journey. I'm your target demographic -- a devops experienced user that is looking for a better CI/CD solution

## Scenario 1 

`dagger init` inside the project directory

```
/my-project
  |- src/
  |- .git/
  |- README.md
  |- LICENSE (or lack thereof)
  |- dagger.json
```

## Scenario 2

`dagger init` above the project directory

```
/dagger-folder
  |- my-project/
    |- src/
    |- .git/
    |- README.md
  |- LICENSE (or lack thereof)
  |- dagger.json
```

## Scenario 3

Dagger module development (I'm not even there yet)

---

I think the license flag is targeting scenario 3, but I'm not sure.  It's not appropriate for new users in either scenario 1 or 2.  Moreover, I don't know how you know the context to distinguish between these different scenarios,  Is `dagger init` appropriate for 1 and 2 and `dagger develop` more for option 3?  Would I ever use `dagger develop` in options 1 and 2?  These are points of confusion regarding dagger module development.

New users will start in scenarios 1 or 2 (I started with scenario 1, which complicated my learning of dagger).  In neither of these cases should a LICENSE ever be created.  In my case, you changed my proprietary code to open-source during `dagger init` -- that's just wrong.  I think @helderco second set of bullets is right on the mark and should be the default.  `--no-license` is too dangerous

**Comment by helderco** [comment]
@drmikecrowe, you're right if I understand you correctly, in that the LICENSE file is intended for the use case of standalone and reusable Dagger modules, that get indexed on the [Daggerverse](https://daggerverse.dev). However, we don't have a good way today to distinguish which case is yours, and the discussion around how to integrate with existing projects is still in flux so things aren't set in stone.
